diff --git a/open_framework.info b/open_framework.info
index e5d8186..8b5a075 100644
--- a/open_framework.info
+++ b/open_framework.info
@@ -6,7 +6,6 @@ version = 2.02
 stylesheets[all][] = bootstrap/css/bootstrap.min.css
 stylesheets[all][] = bootstrap/css/bootstrap-responsive.min.css
 stylesheets[all][] = css/open_framework.css
-scripts[] = js/jquery-1.8.2.min.js
 scripts[] = bootstrap/js/bootstrap.min.js
 scripts[] = js/open_framework.js
 scripts[] = js/override.js
@@ -40,4 +39,4 @@ regions[content_col4-4] = Content: 4-column stacked (4)
 regions[content_lower] = Content: Lower
 regions[content_bottom] = Content: Bottom
 regions[main_lower] = Main: Lower
-regions[main_bottom] = Main: Bottom
\ No newline at end of file
+regions[main_bottom] = Main: Bottom
