diff --git a/panopoly_magic.module b/panopoly_magic.module
index 7fe100c..e5931d2 100644
--- a/panopoly_magic.module
+++ b/panopoly_magic.module
@@ -1215,7 +1215,7 @@ function panopoly_magic_views_pre_render(&$view) {
 
     // Set the view mode
     if (!empty($conf['view_mode'])) {
-      $view->style_plugin->row_plugin->options['view_mode'] = $conf['view_mode'];
+      //$view->style_plugin->row_plugin->options['view_mode'] = $conf['view_mode'];
     }
   }
 }
