diff --git a/panopoly.profile b/panopoly.profile
index 25dad39..d38b439 100755
--- a/panopoly.profile
+++ b/panopoly.profile
@@ -13,7 +13,7 @@ function panopoly_install_tasks(&$install_state) {
 
   // Add the Panopoly app selection to the installation process
   require_once(drupal_get_path('module', 'apps') . '/apps.profile.inc');
-  $tasks = $tasks + apps_profile_install_tasks($install_state, array('machine name' => 'panopoly', 'default apps' => array('panopoly_demo')));
+  $tasks = $tasks + apps_profile_install_tasks($install_state, array('machine name' => 'panopoly', 'default apps' => array()));
 
   // Add the Panopoly theme selection to the installation process
   require_once(drupal_get_path('module', 'panopoly_theme') . '/panopoly_theme.profile.inc');
