diff --git a/index.html b/index.html
deleted file mode 100755
index 98bf81f..0000000
--- a/index.html
+++ /dev/null
@@ -1,189 +0,0 @@
-<!DOCTYPE html>
-<html lang="fr">
-<head>
-    <meta charset="UTF-8">
-    <title>Sharrre.com</title>
-    <meta name="description" content="Sharrre"/>
-    <script src="http://code.jquery.com/jquery-1.7.min.js"></script>
-    <script src="src/js/platform/platform.js"></script>
-    <script src="src/js/platform/twitter.js"></script>
-    <script src="src/js/platform/googlePlus.js"></script>
-    <script src="src/js/platform/facebook.js"></script>
-    <script src="src/js/platform/digg.js"></script>
-    <script src="src/js/platform/delicious.js"></script>
-    <script src="src/js/platform/stumbleupon.js"></script>
-    <script src="src/js/platform/linkedin.js"></script>
-    <script src="src/js/platform/pinterest.js"></script>
-    <script src="src/js/platform/tumblr.js"></script>
-    <script src="src/js/platform/reddit.js"></script>
-    <script src="src/js/platform/twitterFollow.js"></script>
-    <script src="src/js/jquery.sharrre.js"></script>
-    <link rel="stylesheet" href="dist/test.css">
-</head>
-<body>
-
-<script>
-    $(function () {
-        $('h1').each(function () {
-            $this = $(this);
-            $this.append("<a class='show-source'>Show source</a>");
-        });
-
-        $(document).on('click', '.show-source', function () {
-            $this = $(this);
-            $script = $this.parent().prev("script");
-            $this.parent().append("<div class='source'>" + $script.html() + "</div>");
-            $this.remove();
-        });
-    });
-
-</script>
-
-<script>
-    $(function () {
-        $('#demo-all').sharrre({
-            share: {
-                googlePlus: true,
-                facebook: true,
-                twitter: true,
-                delicious: true,
-                stumbleupon: true,
-                linkedin: true,
-                pinterest: true,
-                tumblr: true,
-                reddit: true
-            },
-            buttons: {
-                facebook: {},
-                twitter: {
-                    hashtags: "ShareMe"
-                },
-                delicious: {},
-                linkedin: {},
-                pinterest: {}
-            },
-            hover: function (api, options) {
-                $(api.element).find('.buttons').show();
-            },
-            hide: function (api, options) {
-                $(api.element).find('.buttons').hide();
-            },
-            url: 'http://sharrre.com'
-        });
-    });
-</script>
-<h1>All standard buttons</h1>
-
-<div id="demo-all" class="clearfix"></div>
-
-
-<script>
-    $(function () {
-        $('#demo-no-count').sharrre({
-            share: {
-                googlePlus: true,
-                facebook: true,
-                twitter: true,
-                delicious: true,
-                stumbleupon: true,
-                linkedin: true,
-                pinterest: true,
-                tumblr: true,
-                reddit: true
-            },
-            buttons: {
-                facebook: {},
-                twitter: {},
-                delicious: {},
-                linkedin: {},
-                pinterest: {}
-            },
-            url: 'http://sharrre.com',
-            enableCounter: false
-        });
-    });
-</script>
-<h1>All standard buttons (without counter)</h1>
-
-<div id="demo-no-count" class="clearfix"></div>
-
-<link type="text/css" rel="stylesheet"
-      href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
-<script>
-    var list = ['facebook', 'twitter', 'tumblr', 'reddit', 'twitterFollow', 'googlePlus'];
-    $(function () {
-        $('#demo-custom').sharrre({
-            share: {
-                googlePlus: true,
-                facebook: true,
-                twitter: true,
-                delicious: true,
-                stumbleupon: true,
-                linkedin: true,
-                pinterest: true,
-                tumblr: true,
-                reddit: true,
-                twitterFollow: true
-            },
-            buttons: {
-                facebook: {
-                    popup: {
-                        width: 1200,
-                        height: 900
-                    }
-                },
-                twitter: {},
-                delicious: {},
-                linkedin: {},
-                pinterest: {},
-                twitterFollow: {
-                    user: 'GarethBale11'
-                }
-            },
-            url: 'http://sharrre.com',
-
-            enableCounter: false,
-            enableHover: false,
-            template: function () {
-                content = '';
-                for (i in list) {
-                    service = list[i];
-                    content += '<a class="' + service + '"><i class="fa fa-' + service + '"></i></a>';
-                }
-                return content;
-            }(),
-            render: function (api, options) {
-                for (i in list) {
-                    service = list[i];
-                    $(api.element).on('click', '.' + service, function () {
-                        api.openPopup(this.className);
-                    });
-                }
-            }
-        });
-    });
-</script>
-<h1>Customized buttons</h1>
-
-<div id="demo-custom" class="clearfix"></div>
-
-<script>
-    $(function () {
-        $('#follow').sharrre({
-            share: {
-                twitterFollow: true
-            },
-            buttons: {
-                twitterFollow: {
-                    user: 'GarethBale11'
-                }
-            },
-            url: 'https://github.com'
-        });
-    });
-</script>
-<h1>Twitter follow</h1>
-
-<div id="follow" class="clearfix"></div>
-
-</body>
diff --git a/sharrre.php b/sharrre.php
deleted file mode 100755
index 08a915b..0000000
--- a/sharrre.php
+++ /dev/null
@@ -1,69 +0,0 @@
-<?php
-header('content-type: application/json');
-//Sharrre by Julien Hany
-$json        = array('url' => '', 'count' => 0);
-$json['url'] = $_GET['url'];
-$url         = urlencode($_GET['url']);
-$type        = urlencode($_GET['type']);
-$proxy       = null;
-$proxyAuth   = array();
-if (filter_var($_GET['url'], FILTER_VALIDATE_URL)) {
-    if ($type == 'googlePlus') { //source http://www.helmutgranda.com/2011/11/01/get-a-url-google-count-via-php/
-        $contents = parse('https://plusone.google.com/u/0/_/+1/fastbutton?url=' . $url . '&count=true');
-
-        preg_match('/window\.__SSR = {c: ([\d]+)/', $contents, $matches);
-
-        if (isset($matches[0])) {
-            $json['count'] = (int)str_replace('window.__SSR = {c: ', '', $matches[0]);
-        }
-    } else if ($type == 'stumbleupon') {
-        $content = parse("http://www.stumbleupon.com/services/1.01/badge.getinfo?url=$url");
-
-        $result = json_decode($content);
-        if (isset($result->result->views)) {
-            $json['count'] = $result->result->views;
-        }
-
-    }
-}
-echo str_replace('\\/', '/', json_encode($json));
-
-function parse($encUrl)
-{
-    global $proxy;
-    global $proxyAuth;
-    $options = array(
-        CURLOPT_RETURNTRANSFER => true, // return web page
-        CURLOPT_HEADER         => false, // don't return headers
-        CURLOPT_FOLLOWLOCATION => true, // follow redirects
-        CURLOPT_ENCODING       => "", // handle all encodings
-        CURLOPT_USERAGENT      => 'sharrre', // who am i
-        CURLOPT_AUTOREFERER    => true, // set referer on redirect
-        CURLOPT_CONNECTTIMEOUT => 5, // timeout on connect
-        CURLOPT_TIMEOUT        => 10, // timeout on response
-        CURLOPT_MAXREDIRS      => 3, // stop after 10 redirects
-        CURLOPT_SSL_VERIFYHOST => 0,
-        CURLOPT_SSL_VERIFYPEER => false,
-    );
-    $ch      = curl_init();
-    if ($proxy != null) {
-        $options[CURLOPT_PROXY] = $proxy;
-        if (isset($proxyAuth['user']) && isset($proxyAuth['pwd'])) {
-            $options[CURLOPT_PROXYUSERPWD] = $proxyAuth['user'] . ':' . $proxyAuth['pwd'];
-        }
-    }
-    $options[CURLOPT_URL] = $encUrl;
-    curl_setopt_array($ch, $options);
-
-    $content = curl_exec($ch);
-    $err     = curl_errno($ch);
-    $errmsg  = curl_error($ch);
-
-    curl_close($ch);
-
-    if ($errmsg != '' || $err != '') {
-        /*print_r($errmsg);
-        print_r($errmsg);*/
-    }
-    return $content;
-}
diff --git a/src/js/jquery.sharrre.js b/src/js/jquery.sharrre.js
index 05de60d..941a2ba 100755
--- a/src/js/jquery.sharrre.js
+++ b/src/js/jquery.sharrre.js
@@ -221,7 +221,7 @@
         else { //template by defaults
             $(this.element).html(
                 '<div class="box"><a class="count" href="' + this.options.defaultUrl + '">' + total + '</a>' +
-                (this.options.title !== '' ? '<a class="share" href="' + this.options.defaultUrl + '">' + this.options.title + '</a>' : '') +
+                (this.options.title !== '' ? '<a class="share" href="' + this.options.defaultUrl + '"><span>' + this.options.title + '</span></a>' : '') +
                 '</div>'
             );
         }
@@ -244,15 +244,32 @@
         this.platforms[site].popup(this.options);  //open
         if (this.options.enableTracking === true) { //tracking!
             infos = this.platforms[site].trackingAction;
-            _gaq.push(['_trackSocial', infos.site, infos.action]);
+            //_gaq.push(['_trackSocial', infos.site, infos.action]);
+            //ga('send', 'social', infos.site, infos.action);
+            //ga('send', 'social', {
+              //'socialNetwork': infos.site,
+              //'socialAction': infos.action,
+              //'socialTarget': this.options.url
+            //});
+            dataLayer.push({
+              'socialNetwork': infos.site,
+              'socialAction': infos.action,
+              'socialTarget': this.options.url,
+              'event':'trackSocial'
+            });
         }
     };
 
     /* Methode for add +1 to a counter
      ================================================== */
-    Plugin.prototype.simulateClick = function () {
+    Plugin.prototype.simulateClick = function (className) {
         var html = $(this.element).html();
-        $(this.element).html(html.replace(this.options.total, this.options.total + 1));
+        //$(this.element).html(html.replace(this.options.total, this.options.total + 1));
+        if (className === undefined) {
+          $(this.element).html(html.replace(this.options.total, this.options.total + 1));
+        } else {
+          $('.'+className).html(html.replace(this.options.total, this.options.total + 1));
+        }
     };
 
     /* Methode for add +1 to a counter
diff --git a/src/js/platform/delicious.js b/src/js/platform/delicious.js
index 8ff5c75..76a92b8 100755
--- a/src/js/platform/delicious.js
+++ b/src/js/platform/delicious.js
@@ -16,7 +16,8 @@ SharrrePlatform.register("delicious", function (options) {
         settings: defaultSettings,
         url: function (url) {
             // Doesn't respond on https
-            return 'http://feeds.delicious.com/v2/json/urlinfo/data?url={url}&callback=?';
+            //return 'http://feeds.delicious.com/v2/json/urlinfo/data?url={url}&callback=?';
+            return '//feeds.delicious.com/v2/json/urlinfo/data?url={url}&callback=?';
         },
         trackingAction: {site: 'delicious', action: 'add'},
         load: function (self) {
diff --git a/src/js/platform/digg.js b/src/js/platform/digg.js
index 80fd782..028512b 100755
--- a/src/js/platform/digg.js
+++ b/src/js/platform/digg.js
@@ -18,7 +18,8 @@ SharrrePlatform.register("digg", function (options) {
     return {
         settings: defaultSettings,
         url: function (url) {
-            return "http://services.digg.com/2.0/story.getInfo?links={url}&type=javascript&callback=?";
+            //return "http://services.digg.com/2.0/story.getInfo?links={url}&type=javascript&callback=?";
+            return "//services.digg.com/2.0/story.getInfo?links={url}&type=javascript&callback=?";
         },
         trackingAction: {site: 'digg', action: 'add'},
         load: function (self) {
diff --git a/src/js/platform/facebook.js b/src/js/platform/facebook.js
index 2cdaae6..a668445 100755
--- a/src/js/platform/facebook.js
+++ b/src/js/platform/facebook.js
@@ -24,9 +24,11 @@ SharrrePlatform.register("facebook", function (options) {
     return {
         settings: defaultSettings,
         url: function (url) {
-            return "https://graph.facebook.com/fql?q=SELECT%20url,%20normalized_url,%20share_count,%20like_count,%20comment_count,%20total_count,commentsbox_count,%20comments_fbid,%20click_count%20FROM%20link_stat%20WHERE%20url=%27{url}%27&callback=?";
+            //return "https://graph.facebook.com/fql?q=SELECT%20url,%20normalized_url,%20share_count,%20like_count,%20comment_count,%20total_count,commentsbox_count,%20comments_fbid,%20click_count%20FROM%20link_stat%20WHERE%20url=%27{url}%27&callback=?";
+            return "//graph.facebook.com/fql?q=SELECT%20url,%20normalized_url,%20share_count,%20like_count,%20comment_count,%20total_count,commentsbox_count,%20comments_fbid,%20click_count%20FROM%20link_stat%20WHERE%20url=%27{url}%27&callback=?";
         },
-        trackingAction: {site: 'facebook', action: 'like'},
+        //trackingAction: {site: 'facebook', action: 'like'},
+        trackingAction: {site: 'facebook', action: 'share'},
         load: function (self) {
             var sett = this.settings;
             $(self.element).find('.buttons').append('<div class="button facebook"><div id="fb-root"></div>' +
@@ -66,13 +68,34 @@ SharrrePlatform.register("facebook", function (options) {
             fb = window.setInterval(function () {
                 if (typeof FB !== 'undefined') {
                     FB.Event.subscribe('edge.create', function (targetUrl) {
-                        _gaq.push(['_trackSocial', 'facebook', 'like', targetUrl]);
+                        //_gaq.push(['_trackSocial', 'facebook', 'like', targetUrl]);
+                        //ga('send', 'social', 'facebook', 'like', targetUrl);
+                        dataLayer.push({
+                          'socialNetwork':'facebook',
+                          'socialAction':'like',
+                          'socialTarget':targetUrl,
+                          'event':'trackSocial'
+                        });
                     });
                     FB.Event.subscribe('edge.remove', function (targetUrl) {
-                        _gaq.push(['_trackSocial', 'facebook', 'unlike', targetUrl]);
+                        //_gaq.push(['_trackSocial', 'facebook', 'unlike', targetUrl]);
+                        //ga('send', 'social', 'facebook', 'unlike', targetUrl);
+                        dataLayer.push({
+                          'socialNetwork':'facebook',
+                          'socialAction':'unlike',
+                          'socialTarget':targetUrl,
+                          'event':'trackSocial'
+                        });
                     });
                     FB.Event.subscribe('message.send', function (targetUrl) {
-                        _gaq.push(['_trackSocial', 'facebook', 'send', targetUrl]);
+                        //_gaq.push(['_trackSocial', 'facebook', 'send', targetUrl]);
+                        //ga('send', 'social', 'facebook', 'send', targetUrl);
+                        dataLayer.push({
+                          'socialNetwork':'facebook',
+                          'socialAction':'send',
+                          'socialTarget':targetUrl,
+                          'event':'trackSocial'
+                        });
                     });
                     //console.log('ok');
                     clearInterval(fb);
diff --git a/src/js/platform/linkedin.js b/src/js/platform/linkedin.js
index 13a07fb..d6f74a3 100755
--- a/src/js/platform/linkedin.js
+++ b/src/js/platform/linkedin.js
@@ -14,7 +14,8 @@ SharrrePlatform.register("linkedin", function (options) {
     return {
         settings: defaultSettings,
         url: function (test) {
-            return "https://www.linkedin.com/countserv/count/share?format=jsonp&url={url}&callback=?";
+            //return "https://www.linkedin.com/countserv/count/share?format=jsonp&url={url}&callback=?";
+            return "//www.linkedin.com/countserv/count/share?format=jsonp&url={url}&callback=?";
         },
         trackingAction: {site: 'linkedin', action: 'share'},
         load: function (self) {
@@ -38,7 +39,13 @@ SharrrePlatform.register("linkedin", function (options) {
         },
         tracking: function () {
             function LinkedInShare() {
-                _gaq.push(['_trackSocial', 'linkedin', 'share']);
+                //_gaq.push(['_trackSocial', 'linkedin', 'share']);
+                //ga('send', 'social', 'linkedin', 'share');
+                dataLayer.push({
+                  'socialNetwork':'linkedin',
+                  'socialAction':'share',
+                  'event':'trackSocial'
+                });
             }
         },
         popup: function (opt) {
diff --git a/src/js/platform/pinterest.js b/src/js/platform/pinterest.js
index 327cfa3..81625ac 100755
--- a/src/js/platform/pinterest.js
+++ b/src/js/platform/pinterest.js
@@ -14,7 +14,8 @@ SharrrePlatform.register("pinterest", function (options) {
     return {
         settings: defaultSettings,
         url: function (test) {
-            return "https://api.pinterest.com/v1/urls/count.json?url={url}&callback=?";
+            // return "https://api.pinterest.com/v1/urls/count.json?url={url}&callback=?";
+            return "//api.pinterest.com/v1/urls/count.json?url={url}&callback=?";
         },
         trackingAction: {site: 'pinterest', action: 'pin'},
         load: function (self) {
diff --git a/src/js/platform/twitter.js b/src/js/platform/twitter.js
index 6e14d24..6095fc9 100755
--- a/src/js/platform/twitter.js
+++ b/src/js/platform/twitter.js
@@ -18,7 +18,8 @@ SharrrePlatform.register("twitter", function (options) {
         settings: defaultSettings,
         trackingAction: {site: 'twitter', action: 'tweet'},
         url: function (test) {
-            return "https://opensharecount.com/count.json?url={url}";
+            //return "https://opensharecount.com/count.json?url={url}";
+            return "//opensharecount.com/count.json?url={url}";
         },
         load: function (self) {
             var sett = this.settings;
@@ -49,7 +50,13 @@ SharrrePlatform.register("twitter", function (options) {
                 if (typeof twttr !== 'undefined') {
                     twttr.events.bind('tweet', function (event) {
                         if (event) {
-                            _gaq.push(['_trackSocial', 'twitter', 'tweet']);
+                            //_gaq.push(['_trackSocial', 'twitter', 'tweet']);
+                            //ga('send', 'social', 'twitter', 'tweet');
+                            dataLayer.push({
+                              'socialNetwork':'twitter',
+                              'socialAction':'tweet',
+                              'event':'trackSocial'
+                            });
                         }
                     });
                     clearInterval(tw);
